packages/notifications/src/types.ts

```diff
@@ -38,92 +37,67 @@ export interface NotificationsProvider {
 	// return the name of you provider
 	getProviderName(): string;
 
-	// syncs in-app messages locally
-	syncInAppMessages(): Promise<any>;
+	// get in-app messages from provider
+	getInAppMessages(): Promise<any>;
 
 	// filters in-app messages based on event input and provider logic
-	filterMessages(
+	processInAppMessages(
 		messages: InAppMessage[],
 		event: NotificationEvent
 	): Promise<InAppMessage[]>;
-
-	recordInAppMessageDisplayed(messageId: string): Promise<void>;
 }
 
-export type Layout =
+export type InAppMessageLayout =
 	| 'BOTTOM_BANNER'
 	| 'CAROUSEL'
 	| 'MIDDLE_BANNER'
 	| 'OVERLAYS'
 	| 'TOP_BANNER';
 
-export type ComparisonOperator =
-	| 'EQUAL'
-	| 'GREATER_THAN'
-	| 'GREATER_THAN_OR_EQUAL'
-	| 'LESS_THAN'
-	| 'LESS_THAN_OR_EQUAL';
-
-export type MetricsComparator = (
-	metricsVal: number,
-	eventVal: number
-) => boolean;
-
-type Attribute = {
-	AttributeType: any;
-	Values: string[];
-};
+export type InAppMessageAction = 'CLOSE' | 'DEEP_LINK' | 'LINK';
 
-type EventType = {
-	DimensionType: any;
-	Values: string[];
-};
+interface InAppMessageHeader {
+	content: string;
+	style?: InAppMessageStyle;
+}
 
-type Metric = {
-	ComparisonOperator: ComparisonOperator;
-	Value: number;
-};
+interface InAppMessageBody {
+	content: string;
+	style?: InAppMessageStyle;
+}
 
-type Dimensions = {
-	Attributes?: Record<string, Attribute>;
-	EventType?: EventType;
-	Metrics?: Record<string, Metric>;
-};
+interface InAppMessageImage {
+	src: string;
+}
 
-type EventFilter = {
-	Dimensions: Dimensions;
-	FilterType: any;
-};
+interface InAppMessageButton {
+	title?: string;
+	action?: InAppMessageAction;
```

Comment:
We can make it required, sure. Not sure about the `title` for `content` given that it's a button. With header and body, we kind of know there's intent for potentially rich text content. Do we envision a reason that there'd ever be anything other than the text that's intended for the button here?

Reply:
Potential for a component to be passed as `title`?

Reply:
Isn't this a string coming from the console?

Reply:
In the case of Pinpoint yes, but in the case of other implementations maybe not?

Reply:
Although this whole conversation implies that `string` as a type is not going to work, which i do disagree with. Fine with just keeping it as `title` unless you feel differently

Reply:
Given that we're always going to serialize and store, I'm not sure it's possible. It is possible it could be a stringified JSON that's intended to be consumed by a component maybe.. but even then it's likely to be a title component I think..